### PR TITLE
Add more events and prevent duplicative submission

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Review by hackfisher — APPROVED** [review]

File: pallet/account-migration/src/lib.rs [hunk 1]
```diff
@@ -129,7 +129,9 @@ pub mod pallet {
 	pub enum Event {
 		/// An account has been migrated.
 		Migrated { from: AccountId32, to: AccountId20 },
-		/// An multisig account has been migrated.
+		/// A new multisig account params was noted/recorded on-chain.
+		NewMultisigParamsNoted { from: AccountId32, to: MultisigParams },
```
Comment (boundless-forest) [inline pallet/account-migration/src/lib.rs:133]: Suggest renaming `to: MultisigParams` to `param: MultisigParams`. The struct contain more than just `to` information.

File: pallet/account-migration/src/lib.rs [hunk 2]
```diff
@@ -248,9 +255,13 @@ pub mod pallet {
...
 
-				Self::deposit_event(Event::MultisigMigrated { from, detail });
+				Self::deposit_event(Event::MultisigMigrated { from: from.clone(), detail });
 			} else {
-				<Multisigs<T>>::insert(from, detail);
+				<Multisigs<T>>::insert(&from, detail);
+			}
+
+			if let Some(to) = new_multisig_params {
+				Self::deposit_event(Event::NewMultisigParamsNoted { from, to });
```
Comment (boundless-forest) [inline pallet/account-migration/src/lib.rs:264]: Should we deposit event in the case of `new_multisig_params` is `NONE`? Maybe we should.

Comment (AurevoirXavier) [reply pallet/account-migration/src/lib.rs:264]: Nope

File: pallet/account-migration/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply pallet/account-migration/src/lib.rs:133]: `param` looks confusing to me. The param is all about the `to`, the new multisig.

**Review by boundless-forest — APPROVED** [review]

**Pull request merged**